src/ui/common/src/components/integrations/dialogs/snowflakeDialog.tsx

```diff
@@ -113,6 +114,27 @@ export const SnowflakeDialog: React.FC<Props> = ({
         onChange={(event) => onUpdateField('password', event.target.value)}
         value={value?.password ?? null}
       />
+
+      <IntegrationTextInputField
+        spellCheck={false}
+        required={false}
+        label="Role"
+        description="The role to use when accessing the database above."
+        placeholder={Placeholders.role}
+        onChange={(event) => onUpdateField('role', event.target.value)}
+        value={value?.role ?? null}
```

Comment:
I think you can do just ```value?.role```, unless that returns undefined, I'm actually not 100% sure what value we're getting here.